File: app/src/main/java/com/chattalkie/app/data/remote/api/AuthService.kt
package com.chattalkie.app.data.remote.api

import com.chattalkie.app.data.remote.model.AuthResponse
import com.chattalkie.app.data.remote.model.LoginRequest
import com.chattalkie.app.data.remote.model.RegisterRequest
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthService {
    @POST("api/auth/register")
    suspend fun register(@Body request: RegisterRequest): Response<AuthResponse>

    @POST("api/auth/login")
    suspend fun login(@Body request: LoginRequest): Response<AuthResponse>
}

File: tauri-client/src/services/authService.ts
import { apiClient } from '../api/client';
import { useAuthStore } from '../hooks/useAuth';

// Matches backend AuthResponse
export interface LoginResponse {
    token: string;
    user: {
        id: number;
        username: string;
        name: string;
        avatarUrl?: string;
        status: string;
    };
}

export const authService = {
    login: async (username: string, password: string): Promise<LoginResponse> => {
        const response = await apiClient.post<LoginResponse>('/api/auth/login', { username, password });

        if (response.data.token) {
            useAuthStore.getState().setAuth(
                response.data.token,
                response.data.user.id,
                response.data.user.username,
                response.data.user.name,
                response.data.user.avatarUrl
            );
        }

        return response.data;
    },

    register: async (name: string, username: string, password: string): Promise<LoginResponse> => {
        const response = await apiClient.post<LoginResponse>('/api/auth/register', { name, username, password });
        return response.data;
    }
};
